Integrations/integration-AWS-S3.yml

```diff
@@ -0,0 +1,289 @@
+
+    s3 = boto3.client('s3',aws_access_key_id=AWS_ACCESS_KEY, aws_secret_access_key=AWS_SECRET_KEY)
+
+    def convert_size(size_bytes):
+       if size_bytes == 0:
+           return "0B"
+       size_name = ("B", "KB", "MB", "GB", "TB", "PB", "EB", "ZB", "YB")
+       i = int(math.floor(math.log(size_bytes, 1024)))
+       p = math.pow(1024, i)
+       s = round(size_bytes / p, 2)
+       return "%s %s" % (s, size_name[i])
+
+    def create_entry(name, data, ec):
+        return {
+            'ContentsFormat': formats['json'],
+            'Type': entryTypes['note'],
+            'Contents': data,
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': tableToMarkdown(name, data) if data else 'No result were found',
+            'EntryContext': ec
+        }
+
+
+    def create_bucket(args):
+        data = []
+        response = s3.create_bucket(
+                        ACL=args.get('acl'),
+                        Bucket=args.get('bucketName'),
+                        CreateBucketConfiguration={
+                            'LocationConstraint': args.get('awsRegion')
+                        })
+        data.append ({
+            'BucketName': args.get('bucketName'),
+            'AWSRegion': args.get('awsRegion')
+        })
+
+        ec = {'AWS.S3.NewBucket': data}
+        return create_entry('Created S3 Bucket', data, ec)
+
+    def delete_bucket(args):
+        data =[]
+        response = s3.delete_bucket(Bucket=args.get('bucketName'))
+        data.append ({
+            'BucketName': bucket
+            })
+        ec = {'AWS.S3.DeletedBuckets': data}
+        return create_entry('The following S3 bucket was deleted', data, ec)
+
+    def list_buckets(args):
+        data = []
+        response = s3.list_buckets()
+        for bucket in response['Buckets']:
+            data.append({
+                'BucketName': bucket['Name']
+            })
+
+        ec = {'AWS.S3.BucketList': data}
+        return create_entry('AWS S3 Bucket List', data, ec)
+
+    def get_bucket_policy(args):
+        data = []
+        response = s3.get_bucket_policy(Bucket=args.get('bucketName'))
+        policy = json.loads(response['Policy'])
+        statements = policy['Statement']
+        for statement in statements:
+                data.append({
+                'BucketName': args.get('bucketName'),
+                'PolicyId': policy['Id'],
+                'PolicyVersion': policy['Version'],
+                'Sid': statement['Sid'],
+                'Action': statement['Action'],
+                'Principal': statement['Principal'],
+                'Resource': statement['Resource'],
+                'Effect': statement['Effect']
+                })
+
+        ec = {'AWS.S3.Policy': data, 'AWS.S3.Policy.json': response['Policy']}
+        return create_entry('AWS S3 Bucket Policy', data, ec)
+
+
+    def set_bucket_policy(args):
+        bucket_policy = json.dumps(args.get('bucketPolicyJson'))
```

Comment:
check if `bucketPolicyJson` is dict or string. If dict then dump to json, if string then validate json validity and then pass  to `s3.put_bucket_policy`

Reply:
fixed